src/Engine/ConfigurationState.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\ConfigurationState.
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\Core\Config\Entity\ConfigEntityInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\TypedData\DataDefinitionInterface;
+use Drupal\rules\Entity\ReactionRuleConfig;
+
+/**
+ * The state used during configuration time holding data definitions.
+ */
+class ConfigurationState implements ConfigurationStateInterface {
```

Comment:
I'm not happy with "PreExecutionState" because it sounds like some state which is used before a rule is executed. But we are talking about config time here, which has nothing to do with execution.